Composer/packages/client/src/pages/botProject/adapters/ABSChannels.tsx

```diff
@@ -160,168 +163,115 @@ export const ABSChannels: React.FC<RuntimeSettingsProps> = (props) => {
           loading: false,
         };
       } catch (err) {
-        switch (err?.response.data?.error.code) {
-          case 'ResourceNotFound':
-            // this channel has not yet been created, should display as disabled
-            return {
-              enabled: false,
-              loading: false,
-            };
-            break;
-          case 'AuthenticationFailed':
-            // the auth failed for some reason.
-            break;
-          case 'ResourceGroupNotFound':
-            // this resource group is not found - in other words, can't find a channel registration in the expected spot.
-            break;
-          case 'SubscriptionNotFound':
-            // the subscription is not found or invalid
-            break;
-          default:
-            // handle error.
-            break;
+        if (err?.response.data?.error.code === 'ResourceNotFound') {
+          // this channel has not yet been created, should display as disabled
+          return {
+            enabled: false,
+            loading: false,
+          };
         }
         throw new Error(err?.response.data?.error.message || 'Failed to fetch channel status');
       }
     }
   };
 
-  const createChannelService = async (channelId: string) => {
+  const createChannelService = async (channelId: string, opts?: any) => {
     if (currentResource) {
-      try {
-        const url = `https://management.azure.com/subscriptions/${
-          currentResource.subscriptionId || currentResource.alternateSubscriptionId
-        }/resourceGroups/${currentResource?.resourceGroupName}/providers/Microsoft.BotService/botServices/${
-          currentResource?.resourceName
-        }/channels/${channelId}?api-version=2020-06-02`;
-        let data = {};
-        switch (channelId) {
-          case CHANNELS.TEAMS:
-            data = {
+      const url = `https://management.azure.com/subscriptions/${
+        currentResource.subscriptionId || currentResource.alternateSubscriptionId
+      }/resourceGroups/${currentResource?.resourceGroupName}/providers/Microsoft.BotService/botServices/${
```

Comment:
nit: ```currentResource?.resourceGroupName``` can remove ? because of if expression on L179

Reply:
same goes for all ```currentResource?.whatever``` down below within the if expression block

Reply:
got it